rio_tiler/io/xarray.py

```diff
@@ -0,0 +1,362 @@
+                warnings.warn(
+                    "Cannot determine maxzoom based on dataset information, will default to TMS maxzoom.",
+                    UserWarning,
+                )
+                self._minzoom = self.tms.maxzoom
+
+        return self._minzoom
+
+    def get_maxzoom(self) -> int:
+        """Define dataset maximum zoom level."""
+        if self._maxzoom is None:
+            # TODO: Get transform from the data and figure the MaxZoom
+            # self._minzoom = self.tms.zoom_for_res(resolution)
+            try:
+                dst_affine, _, _ = self._dst_geom_in_tms_crs()
+
+                # The maxzoom is defined by finding the minimum difference between
+                # the raster resolution and the zoom level resolution
+                resolution = max(abs(dst_affine[0]), abs(dst_affine[4]))
+                self._maxzoom = self.tms.zoom_for_res(resolution)
+
+            except:  # noqa
+                # if we can't get min/max zoom from the dataset we default to TMS maxzoom
+                warnings.warn(
+                    "Cannot determine maxzoom based on dataset information, will default to TMS maxzoom.",
+                    UserWarning,
+                )
+                self._maxzoom = self.tms.maxzoom
+
+        return self._maxzoom
+
+    @property
+    def minzoom(self):
+        """Return dataset minzoom."""
+        return self.get_minzoom()
+
+    @property
+    def maxzoom(self):
+        """Return dataset maxzoom."""
+        return self.get_maxzoom()
+
+    def info(self) -> Info:
+        """Return xarray.DataArray info."""
+        dims = [
+            d
+            for d in self.input.dims
+            if d not in ["lat", "lon", "longitude", "latitude", "x", "y"]
+        ]
+
+        bands = [str(band) for d in dims for band in self.input[d].values]
+        metadata = [band.attrs for d in dims for band in self.input[d]]
+
+        meta = {
+            "bounds": self.geographic_bounds,
+            "minzoom": self.minzoom,
+            "maxzoom": self.maxzoom,
+            "band_metadata": [(f"b{ix + 1}", v) for ix, v in enumerate(metadata)],
+            "band_descriptions": [(f"b{ix + 1}", v) for ix, v in enumerate(bands)],
+            "dtype": str(self.input.dtype),
+            "nodata_type": "Nodata" if self.input.rio.nodata is not None else "None",
+            "name": self.input.name,
+            "count": self.input.rio.count,
+            "width": self.input.rio.width,
+            "height": self.input.rio.height,
+        }
+        return Info(**meta)
+
+    def statistics(
+        self,
+        categorical: bool = False,
+        categories: Optional[List[float]] = None,
+        percentiles: List[int] = [2, 98],
+        hist_options: Optional[Dict] = None,
+        max_size: int = 1024,
+        **kwargs: Any,
+    ) -> Dict[str, BandStatistics]:
+        """Return bands statistics from a dataset."""
+        raise NotImplementedError
```

Comment:
I don't think we want to have statistics and preview implemented because it will be really inefficient for now